### ISSUE TITLE
Enforce windows password complexity requirements in acs-engine client. 

### ISSUE THREAD

**Issue by bnookala** [Azure/acs-engine#2407]
**Is this a request for help?**:

No

**Is this an ISSUE or FEATURE REQUEST?** (choose one):

Feature Request

**What version of acs-engine?**:

Any

**Orchestrator and version (e.g. Kubernetes, DC/OS, Swarm)**

Kubernetes, but possibly all orchestrators that support Windows nodes. Any version.

**What happened**:

Deployed a Windows/Linux hybrid cluster, resource group ARM template deployment failed at provisioning the Windows node due to password guidelines not having been met.

**What you expected to happen**:

Do password validation when generating the ARM templates for deployment, to avoid the above deployment failures. The guidelines are here; https://docs.microsoft.com/en-us/windows/security/threat-protection/security-policy-settings/password-must-meet-complexity-requirements.

**How to reproduce it** (as minimally and precisely as possible):

Add a windows agent pool. Set the windows profile adminPassword as simply "password" ie:

```
    "windowsProfile": {
      "adminUsername": "azureuser",
      "adminPassword": "password"
    },
```

**Anything else we need to know**:

Nothing else!

**Comment by CecileRobertMichon** [comment]
@shubheksha this might be a good issue to look at if you're interested

**Comment by CecileRobertMichon** [comment]
@vpatelsj did you make any progress on this?

**Comment by vpatelsj** [comment]
@CecileRobertMichon thanks for reminder, somehow fell off my tracking list somehow. I owe it to the community. Should get a PR by end of the week.

**Comment by tariq1890** [comment]
Closed via #3854
